tsdb/index/postings.go

```diff
@@ -289,41 +289,61 @@ func (p *MemPostings) EnsureOrder(numberOfConcurrentProcesses int) {
 
 // Delete removes all ids in the given map from the postings lists.
 func (p *MemPostings) Delete(deleted map[storage.SeriesRef]struct{}) {
-	var keys, vals []string
+	// Take the optimistic read lock for the entire method,
```

Comment:
Suggestion to explain the whole operation:
```
// Each SeriesRef can appear in several Postings.  To change each one,
// we need to know the label name and value that it is indexed under.
// We iterate over all label names, then for each name all values, and look for
// individual series to be deleted.
```

Reply:
Updated here: https://github.com/prometheus/prometheus/pull/13286/commits/b2dec7dc5f5f24da1970044b23b8333a774c2e95 (modulo my interpretation of semantic line breaks)